### PR TITLE
Unify BlockPreview & BlockPreviewContent into one component

### PR DESCRIPTION
**Pull request by marekhrabe**
## Description

This PR updates the way `BlockPreview` & `BlockPreviewContent` are used. We want to soon make previews available for plugin developers and these two components were quite confusing together as seen in https://github.com/WordPress/gutenberg/pull/16033#issuecomment-499839090. 

This PR introduces a new version of `BlockPreview` which is used for both scaled and full-size previews (controlled by a prop `isScaled`) and allows you to add custom `className` to the preview wrapper for further styling if needed.

`BlockPreviewContent` is gone. 

The boolean prop `isScaled` is a solution that simulates how the component is currently used in the codebase. It will be soon replaced with different way of doing scaling that is being cooked in #16113. For now, I think it's a good solution that will keep our current usages working the same way while paving the way for the new scaling API and exporting the component for devs to use in plugins.

## How has this been tested?
- in Block Inserter - hover any reusable block. A new "panel" with title "Preview" and the preview itself should show up.
- focus some block with styles (button or quote for example):
  - check the block settings sidebar and expand "Block Styles" panel. It should contain previews
  - open the block switcher and check both thumbnails and also the full preview after hover which should match the styling of the one in block inserter

## Screenshots 

This is `<BlockPreview isScaled blocks={ [ … ] } />` inside Block Styles:

Screenshot 2019-07-31 at 11 47 42

This is the same preview inside the block switcher. The "panel" on the right was moved from the `BlockPreview` component directly into places where it is being used from (Inserter and Swicther). BlockPreview component is now always **only** the previewed content and nothing else.

Screenshot 2019-07-31 at 13 53 11

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by marekhrabe** [comment]
TODO: Block Inserter and Block Switcher both use the "preview on hover" fly-out panel but currently don't share CSS for it in my PR. Let's figure out how to do it (without backing that directly into `BlockPreview` like before). 

I'll probably make a wrapper component for `BlockPreview` which will add the "panel" styling and be used it in both places.

**Comment by marekhrabe** [comment]
Rebased against #16033

File: packages/block-editor/src/components/block-preview/index.js [hunk 1]
```diff
@@ -23,22 +24,23 @@ import BlockList from '../block-list';
...
+function BlockPreview( { blocks, settings, className, isScaled } ) {
 	if ( ! blocks ) {
 		return null;
 	}
-
 	return (
-		<Disabled className="editor-block-preview__content block-editor-block-preview__content editor-styles-wrapper" aria-hidden>
+		<Disabled
+			aria-hidden
+			className={ classnames(
```
Comment (retrofox) [inline packages/block-editor/src/components/block-preview/index.js]: minor: not sure about this, but I'm getting an `Irregular whitespace` warning here. May it be a tab?

File: packages/block-editor/src/components/block-preview/index.js [hunk 2]
```diff
@@ -23,22 +24,23 @@ import BlockList from '../block-list';
  *
  * @return {WPElement} Rendered element.
  */
-function BlockPreview( props ) {
-	return (
-		<div className="editor-block-preview block-editor-block-preview">
-			<div className="editor-block-preview__title block-editor-block-preview__title">{ __( 'Preview' ) }</div>
```
Comment (retrofox) [inline packages/block-editor/src/components/block-preview/index.js]: I guess we need to remove the line which imports the `__()` function.

```es6
import { __ } from '@wordpress/i18n';
```

**Comment by marekhrabe** [comment]
I think this is now ready for a review. Components work well for me, as described in the testing instructions.

File: packages/block-editor/src/components/block-preview/index.js [hunk 3]
```diff
@@ -23,22 +23,23 @@ import BlockList from '../block-list';
...
 		return null;
 	}
-
 	return (
-		<Disabled className="editor-block-preview__content block-editor-block-preview__content editor-styles-wrapper" aria-hidden>
+		<Disabled
+			aria-hidden
+			className={
+				classnames(
+					'editor-block-preview', 'block-editor-block-preview',
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-preview/index.js]: I think since we changed the "purpose of the component" anyway, we may want to drop "editor-block-preview"?

File: packages/block-editor/src/components/block-preview/dropdown-wrapper.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * BlockPreviewDropdownWrapper provides styling so you can show Preview inside
+ * Dropdown components. It expects BlockPreview as a single child.
+ *
+ * @param {Object} props Component props.
+ *
+ * @return {WPElement} Rendered element.
+ */
+export default function BlockPreviewDropdownWrapper( { children } ) {
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-preview/dropdown-wrapper.js]: Do we really need this component, or framed differently, how is this related to "preview", it seems like a component used to display something at the right of a popover, right? Should be named something like `PopoverAside` or something like that?

Comment (retrofox) [reply packages/block-editor/src/components/block-preview/dropdown-wrapper.js]: > Should be named something like `PopoverAside` or something like that?

In this case, are you suggesting to put it out of the `<BlockPreview />` component?

Maybe we could try to get the chance to increase its functionality such as `anchorTo: 'top-right`, or even more, make it `autoLocate`: `true`?

For instance, depending on the viewport width the dropdown isn't shown:

image

**It's important to clarify that this wasn't introduced in this PR. It already happens in prod.**

However, creating a new `<PopoverAside />` is out of the scope of this patch imho.

Comment (youknowriad) [reply packages/block-editor/src/components/block-preview/dropdown-wrapper.js]: This PR extracted a new component called `BlockPreviewDropdownWrapper` but if you see its code, there's nothing preview specific aside the label maybe? I agree that it's fine to keep this way. I personally this component is not really needed and should probably be removed and just copy the "preview" title where we need it.

Especially with things like #16813 where this component won't be needed in the inserter anymore.

File: packages/block-editor/src/components/block-switcher/index.js [hunk 5]
```diff
@@ -160,14 +161,15 @@ export class BlockSwitcher extends Component {
...
 						{ ( hoveredClassName !== null ) &&
-							<BlockPreview
-								blocks={
-									cloneBlock( blocks[ 0 ], {
-										className: hoveredClassName,
-									} )
-								}
-
-							/>
+							<BlockPreviewDropdownWrapper>
```
Comment (retrofox) [inline packages/block-editor/src/components/block-switcher/index.js]: Using a `<BlockPreviewDropdownWrapper /> `and `< BlockPreview />` looks slightly complicated unnecessarily. The @youknowriad suggestion of adding a generic `PopoverAside` component could justify this wrapper but as I mentioned above it could require more time and also is kind of out of the scope imo.

What do you think to handle the anchoring inside of the <PreviewBlock /> through of a new property?

We can iterate later if we want to implement the aside mode component.

Comment (youknowriad) [reply packages/block-editor/src/components/block-switcher/index.js]: I  don't think BlockPreview component should concern  itself about anchoring. It should basically be just a simple component that shows a preview of blocks. Any UI outside that (wrapper, popover anchor...) is IMO a separate concern. I don't mind it to be duplicated in the caller if necessary.

Comment (retrofox) [reply packages/block-editor/src/components/block-switcher/index.js]: > It should basically be just a simple component that shows a preview of blocks.

Fair enough.

File: packages/block-editor/src/components/block-preview/dropdown-wrapper.js (hunk 4 above)
Comment (retrofox) [reply packages/block-editor/src/components/block-preview/dropdown-wrapper.js]: > This PR extracted a new component called BlockPreviewDropdownWrapper but if you see its code, there's nothing preview specific aside the label maybe?

For this reason, is what I thought it should be handled by `<BlockPreview />` component.

> I personally this component is not really needed and should probably be removed and just copy the "preview" title where we need it.

... and anchoring the `<PreviewBlock />` component as it currently does? For instance, the `<BlockSwitcher />` takes over this applying styles to the `.block-editor-block-preview` class.

**Comment by retrofox** [comment]
> but I was wondering whether to made it into something more like HOC which will take the same props as BlockPreview and will pass them internally to the preview. The alternative usage could be something like:
> 
> <BlockPreviewForDropdowns blocks={ [ … ] } />

I like the idea of combining the blocks:

```es6
<BlockPreviewDropdownWrapper>
  <BlockPreview blocks={ [ … ] } />
</BlockPreviewDropdownWrapper>
```

However, maybe it worths create the new `<PopoverAside />` component to make it usable not only for the BlockPreview one but other components.

**Comment by retrofox** [comment]
> I made BlockPreviewDropdownWrapper for this. I'm still wondering if it is the right way.

Forgot to mention that if we want to expose the `<BlockPreview />` we will have to expose this one as well, right?

File: packages/block-editor/src/components/block-preview/index.js (hunk 3 above)
Comment (marekhrabe) [reply packages/block-editor/src/components/block-preview/index.js]: Good, I thought it might be the case 👍

**Comment by marekhrabe** [comment]
> I don't mind it to be duplicated in the caller if necessary.

This was my concern and that's why I've introduced the DropdownWrapper to abstract the styling. Knowing this, I've reverted my last commit and we now have a copy of the "aside" in both Inserter and Switcher.

**Review by youknowriad — APPROVED** [review]
I think we may have lost a border around the previews in the change here but tbh I don't think it's worse. It may be better design-wise.

**Comment by shaunandrews** [comment]
> I think we may have lost a border around the previews in the change here but tbh I don't think it's worse. It may be better design-wise.

That was intentional :) Glad you noticed. That border just felt extraneous and was so subtle it was barely noticeable anyways.
